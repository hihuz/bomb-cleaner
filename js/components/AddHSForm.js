import React from 'react';
import { connect } from 'react-redux';
import { addHighScore } from '../actions/actionCreators';

class AddHSForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: ''
    };
    this.updateName = this.updateName.bind(this);
    this.handleAddHSClick = this.handleAddHSClick.bind(this);
  }
  updateName(e) {
    this.setState({ name: e.target.value });
  }
  handleAddHSClick() {
    const date = new Date().toJSON().slice(0, 10);
    this.props.dispatch(addHighScore(
      this.props.mode,
      this.state.name,
      this.props.time,
      date,
      this.props.highScores
    ));
    this.props.closeDialog();
  }
  render() {
    return (
      <div className="hs-form">
        You made a high score for <em>{this.props.mode}</em> mode !
        <br /><br />
        Enter your name below :
        <input type="text" onChange={this.updateName} />
        <button className="hs-validate" onClick={this.handleAddHSClick}>OK</button>
      </div>
    );
  }
}

export default connect()(AddHSForm);
