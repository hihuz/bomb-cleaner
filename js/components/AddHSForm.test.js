import React from 'react';
import { Plain } from './AddHSForm';
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('AddHSForm snapshot', () => {
  const component = shallow(<Plain />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
})
