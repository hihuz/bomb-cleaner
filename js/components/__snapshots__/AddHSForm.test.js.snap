exports[`test AddHSForm snapshot 1`] = `
<div
  className="hs-form">
  Your score is in the top 5 for 
  <em />
   mode !
  <br />
  <br />
  Enter your name below to save it:
  <input
    onChange={[Function]}
    type="text" />
  <button
    className="hs-validate"
    onClick={[Function]}>
    OK
  </button>
</div>
`;
